packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -206,8 +199,12 @@ String? _artifactToFileName(Artifact artifact, Platform hostPlatform, [ BuildMod
       return 'font-subset$exe';
     case Artifact.constFinder:
       return 'const_finder.dart.snapshot';
+    case Artifact.flutterPatchedSdkPath:
+    case Artifact.linuxDesktopPath:
+    case Artifact.windowsDesktopPath:
     case Artifact.flutterToolsFileGenerators:
-      return '';
+      assert(false, 'No filename for a folder path, should not be invoked');
```

Comment:
can we `throw UnimplementedError()`? asserts only run in unit tests, so if we have untested code paths I think it would be better to know about it.